plugins/net_plugin/net_plugin.cpp

```diff
@@ -789,6 +858,108 @@ namespace eosio {
             ( "_lport", local_endpoint_port );
          return mvo;
       }
+
+      // security group management
+      //
+   private:
+      std::optional<chain::account_name> participant_name_;
+      std::atomic<bool> participating_{true};
+
+   public:
+      /** @brief  Returns the optional name of the participant */
+      inline auto participant_name() { return participant_name_; }
+      /** @brief Set the participating status */
+      inline void set_participating(bool status) { 
+         if (my_impl->ssl_enabled)
+            participating_.store(status, std::memory_order_relaxed); 
+      }
+      /** returns true if connection is in the security group */
+      inline bool is_participating() const { 
+         if (my_impl->ssl_enabled)
+            return participating_.load(std::memory_order_relaxed);
+         
+         return true;
+      }
+      inline void setup_participant() {
+         if(my_impl->ssl_enabled) {
+            account_name participant = participant_name_ ? *participant_name_ : account_name{};
```

Comment:
Should just return when participant_name_ is null?

Reply:
no, because you could setup a TLS connected network and never turn on the security_group and we want to keep this connection participating in that case. That will be handled in is_in_security_group(participant) since it checks to see if the security_group is empty, in which case ALL participants are in the security group.